remove `boost::filesystem usage`

psychocoderHPC opened the pull request:
Switch from `boost::filesystem` to `std::filesystem` and reduc ethe boost dependencies.

The second commit is dropping clang tests on ubuntu 18.04 do to libstdc++ incompatibilities between OS shipped g++7 and clang if we compile for C++17.

psychocoderHPC committed:
remove `boost::filesystem usage`

Switch from `boost::filesystem` to `std::filesystem` and reduc ethe
boost dependencies.
- remove boost::filesystem
- remove boost::serialization
- remove boost::system

Add alpakas transitive dependency boost atomic which is used for CPU
accelerators.

steindev commented:
@psychocoderHPC Please note, not only the CI failed, but there are also conflicts to the dev branch. (Maybe because I just merged the "remove boost::math" PR?)

psychocoderHPC commented:
> @psychocoderHPC Please note, not only the CI failed, but there are also conflicts to the dev branch. (Maybe because I just merged the "remove boost::math" PR?)

There are incompatibilities between gnu headers which are too old and C++17. I tried to solve it by using `libc++` but this creates linker issues because of ABI incompatibilities. 
Maybe I will drop ubuntu 18.04 in the CI, it's old anyway and at least with the CI container 3.1 we need to drop it.

psychocoderHPC committed:
CI: drop clang tests on ubuntu 18.04

Ubuntu 18.04 has old g++7 as base compiler, therefore the stdlibc++ does
not have the include `<filesystem>`. Since the clang is using by default
`stdlibc++` we run into compiler issues.
Using `libc++` together with clang creates linker issues with
pre-compiled dependencies.

psychocoderHPC commented:
@steindev PR can be merged, I will not provide the second commit as a separate PR. This will only postpone this PR with very little advantage that we have a PR. We will switch to new CI containers soon and then ubuntu 18.04 will be fully dropped anyways. Is it ok for you?

steindev commented:
FYI, I checked the appearances of `boost::filesystem` w/
```bash
$ grep -rnw include/ -e ".*boost::file.*"
$ #and
$ grep -rnw include/ -e ".*bfs::.*"
```
